test_util/marathon.py

```diff
@@ -202,6 +202,85 @@ def _pool_for_marathon_app(app_id):
             raise Exception("Application deployment failed - operation was not "
                             "completed in {} seconds.".format(timeout))
 
+    def deploy_test_pod_and_check(self, pod, test_uuid):
+        with deploy_pod_and_cleanup(pod) as instances:
+            assert pod['scaling']['instances'] == instances
+
+    def deploy_pod(self, pod_definition, timeout=300, check_health=False):
+        """Deploy a pod to marathon
+
+        This function deploys an a pod and then waits for marathon to
+        aknowledge it's successfull creation or fails the test.
+
+        It waits until all the instances reach tasksRunning and then tasksHealthy state.
+
+        Args:
+            pod_definition: a dict with pod definition as specified in
+                            Marathon API
+            timeout: a time to wait for the pod to reach 'Healthy' status
+                     after which the test should be failed.
+            check_health: wait until Marathon reports tasks as healthy before
+                          returning
+
+        Returns:
+            Scaling instance count
+        """
+        r = self.post('/pods', json=pod_definition)
+        logging.info('Response from marathon: {}'.format(repr(r.json())))
+        assert r.ok
+
+        @retrying.retry(wait_fixed=1000, stop_max_delay=timeout * 1000,
+                        retry_on_result=lambda ret: ret is False,
+                        retry_on_exception=lambda x: False)
+        def _wait_for_pod_deployment(pod_id):
+            r = self.get('/deployments')
+            data = r.json()
+            if len(data) > 0:
+                logging.info('Waiting for pod to be deployed %s', repr(data))
+                return False
+            # deployment complete
+            r = self.get('/pods' + pod_id)
+            data = r.json()
+            return data["scaling"]["instances"]
```

Comment:
Also, I would prefer that we have a return a value similar to what deploy_app does. E.G. can we return all the service points for all the containers? That way I can deploy the dedicated test app inside of a pod and have use most of the same pipes.